### PR TITLE
lib/model: Merge add and start folder funcs and related refactor

### PR DESCRIPTION
**Pull request by imsodin**
Folders are never added and started separately except in tests and the "unlocked" functions aren't used at all except in tests, thus I merged the two into a single function. And I renamed `removeFolderLocked` to `deleteFolderEntriesLocked`, as it isn't the locked equivalent to `removeFolder`. These changes do not change functionality, they just make things more easier to understand and thus more robust (currently one might wonder about the relevance of the distinction of adding and starting a folder, when there isn't any).

For tests that need to inject a custom ignore option and previously did this by adding a folder, then changing ignores and the starting the folder, I just added a variant of `addAndStartFolder` where ignores are passed as an argument.

For send-receive tests that rely on the folder service not actually running and manually constructed a `sendReceiveFolder` object, I changed that to start a model with the desired folder as usual, and then stop the suture services and do any additional changes to the folder object necessary to get tests running. This keeps the setup/settings closer to what is actually happening in production.

And it trims a few lines of code and adds/fixes some test timeouts (which I noticed because I forgot to to reset the folder ctx at first on the send-receive tests, which caused everything to hang).

### CODE REVIEW

File: lib/model/model.go [hunk 1]
```diff
@@ -474,7 +441,7 @@ func (m *model) stopFolder(cfg config.FolderConfiguration, err error) {
 }
 
 // Need to hold lock on m.fmut when calling this.
-func (m *model) removeFolderLocked(cfg config.FolderConfiguration) {
+func (m *model) deleteFolderEntriesLocked(cfg config.FolderConfiguration) {
```
Comment (AudriusButkevicius) [inline lib/model/model.go:444]: Why the rename?

Comment (imsodin) [reply lib/model/model.go:444]: > And I renamed `removeFolderLocked` to `deleteFolderEntriesLocked`, as it isn't the locked equivalent to `removeFolder`.

I consider the name misleading, given `removeFolder` is meant to actually completely remove the folder (as in it's no longer in config and we drop all we know from db), while this is used e.g. when pausing a folder as well.

Comment (AudriusButkevicius) [reply lib/model/model.go:444]: Stop? Cleanup? Shutdown?

Comment (imsodin) [reply lib/model/model.go:444]: `cleanupFolderLocked` should work. Stop is taken and Shutdown not really accurate: It just removes the entries for said folder in all of the maps protected by fmut. That's why I thought a name representing that would be good. So maybe `deleteFolderMapEntriesLocked`? I am fine with either.

**Pull request merged**